src/modules/wallets/sagas.js

```diff
@@ -55,7 +55,30 @@ const generateWallet = function * ({ blockchainKey }) {
   }
 }
 
+const generatePrimaryWallet = function * () {
+  const blockchainKey = blockchains.primary.key
+  const alreadyGenerated = window.localStorage.getItem(`${blockchainKey}-privateKey-current`)
```

Comment:
You should not call something what it is not => will make code impossible to maintain later.

 window.localStorage.getItem(`${blockchainKey}-privateKey-current`) => storedPrivateKey (not alreadyGenerated)
if (!alreadyGenerated) { => if !storedPrivateKey
}

=> that way you read: if no privateKey stored, generate wallet. Which is what this code really does, Naming it alreadyGenerated is misleading.